train_DTI.py

```diff
@@ -183,6 +183,7 @@ def main():
     # Logging
     if "log_file" not in config:
         config.log_file = None
+    os.makedirs(os.path.dirname(config.log_file), exist_ok=True)
```

Comment:
Should only do this if log file is specified. L184/L185 above set `config.log_file` to None if not specified, which would throw an error here trying to get the dirname. 

Also, typically would prefer to use `pathlib.Path` for path operations, something like `log_p = Path(config.log_file); os.makedirs(log_p.parent)`

Reply:
done.